src/consts/Stages.ts

```diff
@@ -184,49 +193,128 @@ export const STAGES: ReadonlyArray<STAGE> = [
+        previous: "DraftPackageHRL",
+        previousEventTitle:
+          "Backward Stage Change: CSF Approval to Draft Package (HRL)",
+        showStage: (request) => request?.csfcaApproval === "Yes",
+      },
+      {
+        key: "SelectionPackageHQApproval",
+        text: "HQ Approval",
+        next: (request) => {
+          if (request?.csfcaApproval === "Yes") {
+            return "SelectionPackageCAApproval";
+          } else {
+            return STAGES.find(({ key }) => key === "PackageApproval")
+              ?.subStages?.find(
+                ({ key }) => key === "SelectionPackageCAApproval"
+              )
+              ?.next?.(request);
+          }
+        },
+        nextEventTitle: (request) => {
+          const next = STAGES.find(({ key }) => key === "PackageApproval")
+            ?.subStages?.find(({ key }) => key === "SelectionPackageHQApproval")
+            ?.next?.(request);
+          return `Forward Stage Change: HQ Approval to ${next}`;
+        },
+        readyForNext: () => true,
+        previous: "DraftPackageHRL",
+        previousEventTitle:
+          "Backward Stage Change: HQ Approval to Draft Package (HRL)",
+        showStage: (request) => request?.hqApproval === "Yes",
+      },
+      {
+        key: "SelectionPackageCAApproval",
+        text: "CA Approval",
+        next: (request) => {
           if (request?.titleV === "Yes") {
-            return "Forward Stage Change: Package Approval to Title V";
+            return "TitleV";
+          } else {
+            return STAGES.find(({ key }) => key === "PackageApproval")
+              ?.subStages?.find(({ key }) => key === "TitleV")
+              ?.next?.(request);
           }
-          return undefined;
+        },
+        nextEventTitle: (request) => {
+          const next = STAGES.find(({ key }) => key === "PackageApproval")
+            ?.subStages?.find(({ key }) => key === "SelectionPackageCAApproval")
+            ?.next?.(request);
+          return `Forward Stage Change: CA Approval to ${next}`;
         },
         readyForNext: () => true,
-        previous: "SelectionPackageOSFApproval",
+        previous: "DraftPackageHRL",
         previousEventTitle:
-          "Backward Stage Change: Package Approval to OSF Approval",
+          "Backward Stage Change: CA Approval to Draft Package (HRL)",
+        showStage: (request) => request?.csfcaApproval === "Yes",
       },
+
       {
         key: "TitleV",
         text: "Title V",
         next: undefined,
         nextEventTitle: undefined,
         readyForNext: () => true,
-        previous: "PackageApproval",
+        previous: "DraftPackageHRL",
         previousEventTitle:
-          "Backward Stage Change: Title V to Package Approval",
+          "Backward Stage Change: Title V to Draft Package (HRL)",
+        showStage: (request) => request?.titleV === "Yes",
       },
     ],
+    showStage: () => true,
   },
   {
     key: "Complete",
```

Comment:
Should we add a "Cancelled" stage?  Or maybe update the RequestsTable so that if it is "Cancelled" it doesn't try resolving what the pretty text for the "Cancelled" key should be, but just returns the pretty text as "Cancelled"?